#!/usr/bin/env python
PACKAGE = "rotate_in_place"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("yaw_goal", double_t, 0, "yaw angle goal relative to starting angle", 0.0, -3.14, 3.14) 
gen.add("kp", double_t, 0, "proportional weight", 0.0, 0.0, 10.0) 
gen.add("ki", double_t, 0, "integral weight", 0.0, 0.0, 10.0) 
gen.add("kd", double_t, 0, "derivative weight", 0.0, 0.0, 10.0) 
gen.add("max_cmd", double_t, 0, "maximum angular velocity", 0.3, 0.0, 1.0) 
gen.add("tolerance", double_t, 0, "goal tolerance", 0.09, 0.0, 0.5) 
gen.add("timeout", double_t, 0, "controller timeout", 10.0, 0.0, 60.0) 
gen.add("controller_frequency", double_t, 0, "controller frequency", 10.0, 1.0, 100.0) 
gen.add("projected_pose_angle", double_t, 0, "yaw angle of a projected pose for future safety check", 0.0, 0.0, 1.57) 

exit(gen.generate(PACKAGE, "rotate_in_place", "RotateInPlace"))
